llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -627,11 +627,12 @@ static bool removeSYCLKernelsConstRefArray(Module &M) {
          "Cannot remove initializer of llvm.used global");
   Initializer->destroyConstant();
   for (auto It = IOperands.begin(); It != IOperands.end(); It++) {
-    auto Op = (*It)->getOperand(0);
+    auto Op = (*It)->stripPointerCasts();
```

Comment:
Will this work for non-opaque pointer as well?

Reply:
Yes. In the typed pointer case, you'd expect to see `bitcast void ()* @foo to i8*` and `->stripPointerCasts()` will return the function `@foo` in that case. There are existing tests for sycl-post-link that support the non-opaque case that ensures this code remains correct (I copied them and converted them to opaque pointers for the new tests here).